runtime/src/bank.rs

```diff
@@ -1319,45 +1332,54 @@ impl Bank {
                     // needed for timestamp bounding, but isn't yet corrected for the activation slot
                     let epoch_start_timestamp = if self.slot() > timestamp_bounding_activation_slot
                     {
-                        let epoch = if let Some(epoch) = parent_epoch {
-                            epoch
+                        let warp_testnet_timestamp = self
+                            .feature_set
+                            .activated_slot(&feature_set::warp_testnet_timestamp::id());
+                        if warp_testnet_timestamp.is_some()
```

Comment:
nits: I think this can just be `if warp_testnet_timestamp == Some(self.slot())` with simpler condition and fewer `.unwrap()`.